src/components/input-fields/with-label/WithLabel.tsx

```diff
@@ -0,0 +1,51 @@
+import React from "react";
+import { WithAddon } from "../with-addon/WithAddon";
+import { SLabel } from "./styles/SLabel";
+
+interface ILabel {
+  content: string;
+  active?: boolean;
+}
+
+export const WithLabel: React.FC<ILabel> = ({ ...props }) => {
+  return (
+    <WithAddon
+      name="label"
+      component={<SLabel>{props.content}</SLabel>}
+      content={props.content}
+      active={props.active}
+    />
+  );
+};
+
+interface IWithLabelProps {
+  label: string;
+  addons?: any[];
+}
+
+interface IAddonHOC {
+  addons?: any[];
+}
+
+export function withLabel<T extends IAddonHOC>(WrappedComponent: React.ComponentType<T>) {
+  const displayName = WrappedComponent.displayName || WrappedComponent.name || "Component";
+
+  const WithLabel: React.ComponentType<T & IWithLabelProps & IAddonHOC> = (props) => {
+    const newAddon = <SLabel>{props.label}</SLabel>;
+
+    const addons = props?.addons?.slice() || [];
+    addons.push(newAddon);
```

Comment:
@nicolasshiken I meant this to be internal state of this component, props are ready only. I meant to copy and add this into internal state and handle the registration on componentDidMount only with an effect.

Reply:
Do you mean something like this?
<img width="479" alt="Captura de Pantalla 2021-11-19 a la(s) 17 48 53" src="https://user-images.githubusercontent.com/69903081/142689920-8d8a27c6-2c5a-4b84-a550-fdc99b279d57.png">